src/compiler.jl

```diff
@@ -3552,14 +3552,16 @@ function GPUCompiler.codegen(
 
         caller = mi
         if mode == API.DEM_ForwardMode
-            has_custom_rule =
-                EnzymeRules.has_frule_from_sig(specTypes; world, method_table, caller)
+            # This passes nothing which doesn't add an invalidation, but this
+            # is already handled via absint to get here
+            has_custom_rule = EnzymeRules.has_frule_from_sig(interp, specTypes, nothing)
```

Comment:
This should just use the old code with caller=nothing

Reply:
Is there a caching or other advantage to using the interpreter here since it will be the same as inference?

Reply:
Not really, the key is SV and us batching the edge update